### PR TITLE
[BEAM-14065] Upgrade vendored bytebuddy to version 1.12.8

### PR DESCRIPTION
**Pull request by cushon**
R: @kennknowles

### CODE REVIEW

**Comment by cushon** [comment]
The CI failures look possibly like flakes:

```
java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1469)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1498)
	at java.util.AbstractCollection.toString(AbstractCollection.java:461)
	at java.lang.String.valueOf(String.java:2994)
	at java.lang.StringBuilder.append(StringBuilder.java:131)
	at org.apache.beam.sdk.transforms.ParDoLifecycleTest$ExceptionThrowingFn.validate(ParDoLifecycleTest.java:392)
```

in `org.apache.beam.sdk.transforms.ParDoLifecycleTest.testTeardownCalledAfterExceptionInFinishBundleStateful`: https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/21347/testReport/org.apache.beam.sdk.transforms/ParDoLifecycleTest/testTeardownCalledAfterExceptionInFinishBundleStateful/

And:

```
09:23:46 > Task :beam-test-infra-metrics:validateConfiguration
09:23:46 W0307 17:23:44.399526 3830908 helpers.go:549] --dry-run=true is deprecated (boolean value) and can be replaced with --dry-run=client.
09:24:15 Unable to connect to the server: dial tcp 104.154.102.21:443: i/o timeout
```

in https://ci-beam.apache.org/job/beam_PreCommit_CommunityMetrics_Commit/5078/console

File: vendor/bytebuddy-1_12_8/build.gradle.kts [hunk 1]
```diff
@@ -28,12 +28,12 @@ vendorJava(
   mapOf(
     "dependencies" to listOf("net.bytebuddy:byte-buddy:1.11.0"),
     "relocations" to mapOf(
-            "net.bytebuddy" to "org.apache.beam.vendor.bytebuddy.v1_11_0.net.bytebuddy"),
+            "net.bytebuddy" to "org.apache.beam.vendor.bytebuddy.v1_12_8.net.bytebuddy"),
```
Comment (kennknowles) [inline vendor/bytebuddy-1_12_8/build.gradle.kts:31]: Lines 21 (`description` field) and line 29 (the actual bytebuddy version that gets downloaded and re-namespaced) each need an update

**Review by kennknowles — COMMENTED** [review]
Thanks for doing this!

File: vendor/bytebuddy-1_12_8/build.gradle.kts (hunk 1 above)
Comment (cushon) [reply vendor/bytebuddy-1_12_8/build.gradle.kts:31]: Fixed, thanks!

**Comment by aaltay** [comment]
What is next on this PR?

**Comment by cushon** [comment]
@aaltay I think I've addressed all of the feedback from the initial round of review, and that it's ready for another pass

**Comment by aaltay** [comment]
@kennknowles - PTAL?

**Review by kennknowles — APPROVED** [review]

**Pull request merged**